lib/image_stack.dart

```diff
@@ -48,52 +128,110 @@ class ImageStack extends StatelessWidget {
           .sublist(1, _size)
           .asMap()
           .map((index, image) => MapEntry(
-                index,
-                Positioned(
-                  right: 0.8 * imageRadius * (index + 1.0),
-                  child: circularImage(image),
-                ),
-              ))
+        index,
+        Positioned(
+          right: 0.8 * imageRadius * (index + 1.0),
+          child: circularImage(image),
+        ),
+      ))
+          .values
+          .toList());
+    }
+    if (children.length > 1) {
+      if (children.length < _size) {
+        _size = children.length;
+      }
+      widgets.addAll(children
+          .sublist(1, _size)
+          .asMap()
+          .map((index, widget) => MapEntry(
+        index,
+        Positioned(
+          right: 0.8 * widgetRadius * (index + 1.0),
+          child: circularWidget(widget),
+        ),
+      ))
+          .values
+          .toList());
+    }
+    if (providers.length > 1) {
+      if (providers.length < _size) {
+        _size = providers.length;
+      }
+      providersImages.addAll(providers
+          .sublist(1, _size)
+          .asMap()
+          .map((index, data) => MapEntry(
+        index,
+        Positioned(
+          right: 0.8 * imageRadius * (index + 1.0),
+          child: circularProviders(data),
+        ),
+      ))
           .values
           .toList());
     }
     return Container(
       child: Row(
         children: <Widget>[
-          images.isNotEmpty
+          images.isNotEmpty || widgets.isNotEmpty || providersImages.isNotEmpty
               ? Stack(
-                  overflow: Overflow.visible,
-                  textDirection: TextDirection.rtl,
-                  children: images,
-                )
+            overflow: Overflow.visible,
+            textDirection: TextDirection.rtl,
+            children: children.length > 0
+                ? widgets
+                : providers.length > 0 ? providersImages : images,
+          )
               : SizedBox(),
           Container(
             margin: EdgeInsets.only(left: 5),
             child: totalCount - images.length > 0
```

Comment:
images.length will always take the images list and not widgets or providersImages to get the remaining count to show.

```suggestion
            child: totalCount - (check for all the list's length to get the remaining count) > 0
```